ddcz/templatetags/creations.py

```diff
@@ -47,6 +48,11 @@ def creation_canonical_url(page, creation):
     return page.get_creation_canonical_url(creation)
 
 
+@register.filter
+def singular_name_from_slug(slug):
+    return CreativePage.objects.get(slug=slug).singular_name
```

Comment:
This is unnecessary additional query since you can pass a creative page from view where you already have it